### ISSUE TITLE
Assert failure: Consistency check failed: AV in clr at this callstack

### ISSUE THREAD

**Issue by VincentBu** [dotnet/coreclr#25049]
**Job:**
https://mc.dot.net/#/user/coreclr-outerloop-gcstress-extra/ci~2Fdotnet~2Fcoreclr~2Frefs~2Fheads~2Fmaster/test~2Ffunctional~2Fcli~2F/20190608.1/workItem/JIT/analysis/xunit/JIT_Performance._CodeQuality_BenchmarksGame_regex_redux_regex_redux_5_regex_redux_5_~2F_CodeQuality_BenchmarksGame_regex_redux_regex_redux_5_regex_redux_5_cmd

**Failed tests:**
JIT_Performance._CodeQuality_BenchmarksGame_regex_redux_regex_redux_5_regex_redux_5_._CodeQuality_BenchmarksGame_regex_redux_regex_redux_5_regex_redux_5_cmd

**Log:**
```
Assert failure(PID 3820 [0x00000eec], Thread: 836 [0x0344]): Consistency check failed: AV in clr at this callstack:
------
CORECLR! GetCLRRuntimeHost + 0x5C3910 (0x7313f330)
<no module>! <no symbol> + 0x0 (0x04e6b33b)
<no module>! <no symbol> + 0x0 (0x04e6b2ce)
<no module>! <no symbol> + 0x0 (0x08f24deb)
<no module>! <no symbol> + 0x0 (0x08f24d18)
<no module>! <no symbol> + 0x0 (0x08f242cd)
<no module>! <no symbol> + 0x0 (0x082dedae)
<no module>! <no symbol> + 0x0 (0x079ff540)
<no module>! <no symbol> + 0x0 (0x082da446)
<no module>! <no symbol> + 0x0 (0x082da3c6)
<no module>! <no symbol> + 0x0 (0x08f23523)
<no module>! <no symbol> + 0x0 (0x082da1c2)
<no module>! <no symbol> + 0x0 (0x082d9b3e)
<no module>! <no symbol> + 0x0 (0x082d97f2)
<no module>! <no symbol> + 0x0 (0x082d9404)
<no module>! <no symbol> + 0x0 (0x082d932b)
<no module>! <no symbol> + 0x0 (0x082d6bfb)
CORECLR! GetCLRRuntimeHost + 0x266BB9 (0x72de25d9)
CORECLR! GetCLRRuntimeHost + 0x266D36 (0x72de2756)
-----
.AV on tid=0x344 (836), cxr=0910E3F0, exr=0910E3A0
FAILED: false

CORECLR! coreclr_shutdown_2 + 0xAADA (0x72b4828a)
CORECLR! GetCLRRuntimeHost + 0x2D64E (0x72ba906e)
CORECLR! GetCLRRuntimeHost + 0x2D2B4 (0x72ba8cd4)
CORECLR! GetCLRRuntimeHost + 0x2D25E (0x72ba8c7e)
CORECLR! GetCLRRuntimeHost + 0x2D888 (0x72ba92a8)
NTDLL! LdrSetAppCompatDllRedirectionCallback + 0x1CB63 (0x779304c3)
NTDLL! RtlUnwind + 0x1BA (0x778f051a)
NTDLL! KiUserExceptionDispatcher + 0xF (0x77900fff)
<no module>! <no symbol> + 0x0 (0x04e6b33b)
<no module>! <no symbol> + 0x0 (0x04e6b2ce)
    File: f:\\workspace\\_work\\1\\s\\src\\vm\\excep.cpp Line: 7601
    Image: C:\\dotnetbuild\\work\\9b9a56fc-7ac1-4b06-9241-faeab9e86a1c\\Payload\\CoreRun.exe


Return code:      1
Raw output file:      C:\\dotnetbuild\\work\\9b9a56fc-7ac1-4b06-9241-faeab9e86a1c\\Work\\85330283-2dc8-4728-8b9e-6f908e4b4b25\\Exec\\JIT\\Performance\\Reports\\JIT.Performance\\CodeQuality\\BenchmarksGame\\regex-redux\\regex-redux-5\\regex-redux-5.output.txt
Raw output:
BEGIN EXECUTION
 \"C:\\dotnetbuild\\work\\9b9a56fc-7ac1-4b06-9241-faeab9e86a1c\\Payload\\corerun.exe\" regex-redux-5.exe 
Expected: 100
Actual: -1073740286
END EXECUTION - FAILED
FAILED
Test Harness Exitcode is : 1
To run the test:
> set CORE_ROOT=C:\\dotnetbuild\\work\\9b9a56fc-7ac1-4b06-9241-faeab9e86a1c\\Payload
> C:\\dotnetbuild\\work\\9b9a56fc-7ac1-4b06-9241-faeab9e86a1c\\Work\\85330283-2dc8-4728-8b9e-6f908e4b4b25\\Exec\\JIT\\Performance\\CodeQuality\\BenchmarksGame\\regex-redux\\regex-redux-5\\regex-redux-5.cmd
Expected: True
Actual:   False
```

**Comment by BruceForstall** [comment]
@sergiy-k

**Comment by BruceForstall** [comment]
@echesakovMSFT @RussKeldorph @jashook It looks like the VM has no info about where the PDB files are, so can't construct a good callstack. Do the PDBs get copied to the Helix machines? If they're not in the same directory as the DLLs, we need to set `_NT_SYMBOL_PATH` to include them. (Not sure about non-Windows.)

**Comment by RussKeldorph** [comment]
@mikem8361 @janvorli Any symbol recommendations?

**Comment by mikem8361** [comment]
I don't really have anymore more info other than Bruce is right that the PDBs need to be next to the DLLs.  I'm assuming that is the runtime (coreclr) assert code (checked runtime build?) displaying the above stack.

**Comment by RussKeldorph** [comment]
Assigning to briansull for now since this could be related to #25244.

**Comment by briansull** [comment]
ran GCSTRESS=C in a loop on
JIT\Performance\CodeQuality\BenchmarksGame\regex-redux\regex-redux-5\regex-redux-5.exe

And saw one failure that looked like a GC hole:

Unhandled exception. System.AggregateException: One or more errors occurred. (Object reference not set to an instance of an object.)
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at System.Reflection.Emit.ILGenerator.Emit(OpCode opcode, Label label)
   at System.Text.RegularExpressions.RegexCompiler.BleFar(Label l) in /_/src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs:line 548
   at System.Text.RegularExpressions.RegexCompiler.GenerateFindFirstChar() in /_/src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs:line 1465
   at System.Text.RegularExpressions.RegexLWCGCompiler.FactoryInstanceFromCode(RegexCode code, RegexOptions options) in /_/src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexLWCGCompiler.cs:line 39
   at System.Text.RegularExpressions.RegexCompiler.Compile(RegexCode code, RegexOptions options) in /_/src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs:line 171
   at System.Text.RegularExpressions.Regex.Compile(RegexCode code, RegexOptions roptions) in /_/src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs:line 213
   at System.Text.RegularExpressions.Regex..ctor(String pattern, RegexOptions options, TimeSpan matchTimeout, Boolean addToCache) in /_/src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs:line 160
   at System.Text.RegularExpressions.Regex..ctor(String pattern, RegexOptions options) in /_/src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs:line 61

**Comment by briansull** [comment]
Running GCStress=C under windbg in a loop, hit this assert:

```
inline unsigned SKIP_MOV_REG_REG(PTR_CBYTE base, unsigned offset)
{
    LIMITED_METHOD_DAC_CONTRACT;

    // Confirm it is a move instruction
    // Note that only the first byte may have been stomped on by IsMarkerInstr()
    // So we can check the second byte directly
    _ASSERTE(CheckInstrBytePattern(base[offset] & 0xFD, 0x89, base[offset]) &&
             (base[offset+1] & 0xC0) == 0xC0);
    return(offset + 2);
}
```

The instruction code is `**0xfb**` which isn't one of the expected values:

...
bool IsMarkerInstr(BYTE val)
{
    SUPPORTS_DAC; 
#ifdef _DEBUG
    return (val == X86_INSTR_INT3) || // Debugger might stomp with an int3
           (val == X86_INSTR_HLT && GCStress<cfg_any>::IsEnabled()); // GcCover might stomp with a Hlt
#else
    return false;
#endif
}

0:008> !U 0x08c49a2e
Normal JIT generated code
System.Text.RegularExpressions.RegexCompiler.DeclareString()
Begin 08c499e8, size 4a

/_/src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs @ 1604:
08c499e8 55              push    ebp
08c499e9 8bec            mov     ebp,esp
08c499eb 83ec10          sub     esp,10h
08c499ee 33c0            xor     eax,eax
08c499f0 8945f8          mov     dword ptr [ebp-8],eax
08c499f3 8945f4          mov     dword ptr [ebp-0Ch],eax
08c499f6 8945f0          mov     dword ptr [ebp-10h],eax
08c499f9 894dfc          mov     dword ptr [ebp-4],ecx
08c499fc 8b4dfc          mov     ecx,dword ptr [ebp-4]
08c499ff 8b4904          mov     ecx,dword ptr [ecx+4]
08c49a02 894df4          mov     dword ptr [ebp-0Ch],ecx
08c49a05 b9a07c5005      mov     ecx,5507CA0h (MT: System.String)
08c49a0a e8917a0872      call    CoreCLR!JIT_GetRuntimeType (7acd14a0)
08c49a0f 8945f8          mov     dword ptr [ebp-8],eax
08c49a12 ff75f8          push    dword ptr [ebp-8]
08c49a15 e846f13b72      call    CoreCLR!RuntimeTypeHandle::GetTypeFromHandle (7b008b60)
08c49a1a 8945f0          mov     dword ptr [ebp-10h],eax
08c49a1d 8b55f0          mov     edx,dword ptr [ebp-10h]
08c49a20 8b4df4          mov     ecx,dword ptr [ebp-0Ch]
08c49a23 8b45f4          mov     eax,dword ptr [ebp-0Ch]
08c49a26 8b00            mov     eax,dword ptr [eax]
08c49a28 8b403c          mov     eax,dword ptr [eax+3Ch]
08c49a2b ff5014          call    dword ptr [eax+14h]
>>> 08c49a2e 8be5            mov     esp,ebp (gcstress)
08c49a30 5d              pop     ebp
08c49a31 c3              ret (gcstress)
0:008> u 0x08c49a2e
system_text_regularexpressions!System.Text.RegularExpressions.RegexCompiler.DeclareString()+0x46 [/_/src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs @ 1604]:
08c49a2e fb              sti
08c49a2f e55d            in      eax,5Dh
08c49a31 f4              hlt
...

**Comment by briansull** [comment]
Resolved by #25445
